graphql_compiler/compiler/directive_helpers.py

```diff
@@ -179,3 +179,29 @@ def validate_vertex_field_directive_interactions(location, directives):
     if optional_directive and recurse_directive:
         raise GraphQLCompilationError(u'@optional and @recurse may not appear at the same '
                                       u'vertex field! Location: {}'.format(location))
+
+
+def validate_vertex_field_directive_in_context(location, directives, context):
+    """Ensure that the specified vertex field directives are allowed in the current context."""
+    fold_directive = directives.get('fold', None)
+    optional_directive = directives.get('optional', None)
+    output_source_directive = directives.get('output_source', None)
+    recurse_directive = directives.get('recurse', None)
+
+    fold_context = 'fold' in context
+    # optional_context = 'optional' in context
+    # output_context = 'output' in context
+    # recurse_context = 'recurse' in context
```

Comment:
commented out code?

Reply:
It's outdated, @obi1kenobi - I already flagged this.